src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs

```diff
@@ -142,6 +143,31 @@ public IModelMetadataProvider MetadataProvider
         }
 
         /// <summary>
+        /// Gets or sets the <see cref="IModelBinderFactory"/>.
+        /// </summary>
+        public IModelBinderFactory ModelBinderFactory
```

Comment:
Every service dependency on `Controller` works like this.

> Note this does not support the same scenarios as changing the IModelBinders in Controller.ModelBinders would do w/ existing code -- at least not in a straightforward fashion.

Not intended to be supported. 

This property was always 'unit testing only' in legacy code.